docs/kbs_attestation_protocol.md

```diff
@@ -1,220 +1,418 @@
+Encryption algorithm used to encrypt the output of the KBS service API.
 
-- `tee-pubkey`
+- `ciphertext`
 
-After KBC receives the attestation challenge, an ephemeral asymmetric key pair is generated 
-in HW-TEE. The private key is stored in HW-TEE. The public key and its description information 
-are exported and placed in the `tee-pubkey` field and sent to KBS together with evidence. The 
-hash of the `tee-pubkey` field needs to be included in the custom field of HW-TEE evidence and 
-signed by HW-TEE hardware. This public key is valid until the next time KBC receives this KBS's 
-attestation challenge.
+The output of the KBS service API. It must be encrypted with the KBS-generated
+ephemeral key.
 
-- `tee-evidence`
+- `iv`
 
-The specific content of evidence related to the HW-TEE platform software and hardware in KBC's 
-environment, different `tee-evidence` formats will be defined according to the type of TEE. This 
-format is defined by the Attestation-Service. 
-**KBS will not analyze the content of this structure, but will directly forward it to the Attestation-Service for verification**.
+The input to a cryptographic primitive being used to provide the initial state.
+If the algorithm described by `enc` used does not need it, this field is left
+blank.
 
-## Response
+- `encrypted_key`
 
-If Attestation-Service fails to verify the evidence, KBS returns error response information to 
-KBC.
+The encrypted symmetric key used to encrypt `ciphertext`.
+This key is encrypted with the HW-TEE's public key, using the algorithm defined
+in `alg`.
 
-If KBS verified the token or the attestation evidence successfully, it will return a response to KBC in the following format:
+## Key Format
 
-```json
+### Public Key
+
+The [`Attestation`](#attestation) and the [Token Resource](#token-resource)
+`tee-pubkey` field contains a public key from a HW-TEE generated asymmetric
+key pair.
+This field follow the [JSON Web Key](https://www.rfc-editor.org/rfc/rfc7517)
+format:
+
+``` json
 {
-    "response": {
-        /* The output of KBS service API, needs to be encrypted by a symmetric key randomly generated by KBS. */
-        "output": "",
-        /* Symmetric key and algorithm information for encrypting `output`. */
-        "crypto-annotation": {
-            "algorithm": "",
-            /* The input to a cryptographic primitive being used to provide the initial state, if the algorithm used does not need it, this field left blank */
-            "initialization-vector": "",
-            /* The symmetric key used to encrypt `output`, which is encrypted by HW-TEE's public key */
-            "enc-symkey": ""
-        },
-        /* The token issued by KBS to KBC. If there is a token in the request, this field will be left blank. */
-        "token": "",
-    }
+    "kty": "$key_type",
+    "alg": "$key_algorithm"
+    "k": "public_key"
 }
 ```
 
-- `output`
+# HTTP Integration
+
+KBS uses the HTTP transport protocol to exchange the above described
```

Comment:
This is rather confusing as the introduction says we use HTTPS (line 27).

Reply:
Fixed, thanks.